sys/hashes/psa_riot_hashes/sha_3.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2024 TU Dresden
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_psa_crypto
+ * @{
+ *
+ * @brief       Glue code translating between PSA Crypto and the RIOT SHA3 Hash module
+ *
+ * @author      Lennard Melling <lennard.melling@msx.tu-dresden.de>
+ *
+ * @}
+ */
+
+#include "psa/crypto.h"
+#include "hashes/psa/riot_hashes.h"
+
+psa_status_t psa_hashes_sha3_update(psa_hashes_sha3_ctx_t *ctx,
+                                      const uint8_t *input,
+                                      size_t input_length)
+{
+    sha3_update((keccak_state_t *)ctx, input, input_length);
+    return PSA_SUCCESS;
```

Comment:
Unfortunate that all of those functions don't return an error code, but it is what it is :)

Reply:
That's true! Maybe the underlying functions will return something in the future, then the glue-code can also return different error codes :)